webportal/src/app/job/job-submit/job-submit.component.js

```diff
@@ -65,8 +74,12 @@ $(document).ready(() => {
   $(document).on('change', '#file', (event) => {
     const reader = new FileReader();
     reader.onload = (event) => {
-      const jobConfig = JSON.parse(event.target.result);
-      submitJob(jobConfig);
+      const jobConfig = event.target.result;
+      if (isValidJson(jobConfig)) {
+        submitJob(JSON.parse(jobConfig));
+      } else {
+        alert('Please upload a valid json file.');
```

Comment:
>alert('Please upload a valid json file.'); [](start = 8, length = 42)

Alert the exception information will more helpful for user to detect the Json format issue

Reply:
The js SyntaxError doesn't provide much information and I think print error stack in production env is not a good idea.

Reply:
You can show the alert and attach original exception somewhere for interested users (like us)

---
In reply to: [159381548](https://github.com/Microsoft/pai/pull/162#discussion_r159381548) [](ancestors = 159381548)

Reply:
updated